doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        some special attributes on estimator objects:
+
+        ``_estimator_type``
+            This string-valued attribute identifies an estimator as being a
+            classifier, regressor, etc. It is set by mixins such as
+            :class:`base.ClassifierMixin`, but needs to be more explicitly
+            adopted on a :term:`meta-estimator`.  Its value should usually be
+            checked by way of a helper such as :func:`base.is_classifier`.
+
+        ``_pairwise``
+            This boolean attribute indicates whether the data (``X``) passed to
+            :func:`fit` and similar methods consists of pairwise measures over
+            samples rather than a feature representation for each sample.  It
+            is usually ``True`` where an estimator has a ``metric`` or
+            ``affinity`` or ``kernel`` parameter with value 'precomputed'.
+
+    feature
+    features
+        A feature is a function mapping a sampled object to a numeric or
+        categorical quantity.  "Feature" is also commonly used to refer to
+        these quantities, being the individual elements of a vector
+        representing a sample. In a data matrix, features are represented as
+        columns: each column contains the result of applying a feature
+        function to a set of samples.
+
+        Elsewhere features are known as attributes, predictors, regressors, or
+        independent variables.
+
+        Features in scikit-learn are generally expected to be numeric and
+        finite, even when they have semantically distinct domains and
+        distributions (categorical, ordinal, count-valued, real-valued,
+        interval). See also :term:`categorical feature` and :term:`missing
+        values`.
+
+        ``n_features`` indicates the number of features in a dataset.
+
+    fitting
+        Calling :term:`fit` (or :term:`fit_transform`, :term:`fit_predict`,
+        etc.) on an estimator.
+
+    fitted
+        The state of an estimator after :term:`fitting`.
+
+    function
+        TODO
+
+        Talk about where estimator fit or fit_transform functionality is present in a function.
+
+    joblib
+        A Python library (http://joblib.readthedocs.io) used in Scikit-learn to
+        facilite simple parallelism and caching.  Joblib is oriented towards
+        efficiently working with numpy arrays, such as through use of
+        :term:`memory mapping`.
+
+    leakage
+    data leakage
+        A problem in cross validation where generalization performance can be
+        over-estimated since knowledge of the test data was inadvertently
+        included in training a model.  This is a risk, for instance, when
+        applying a :term:`transformer` to the entirety of a dataset rather
+        than each training portion in a cross validation split.
+
+        We aim to provide interfaces (such as :mod:`pipeline` and
+        :mod:`model_selection`) that shield the user from data leakage.
+
```

Comment:
meta-estimator?

Reply:
Listed under "estimator types" below, even if perhaps it shouldn't be.